registry/handlers/images.go

```diff
@@ -179,8 +172,7 @@ func (imh *imageManifestHandler) DeleteImageManifest(w http.ResponseWriter, r *h
 	// tag index entries a serious problem in eventually consistent storage.
 	// Once we work out schema version 2, the full deletion system will be
 	// worked out and we can add support back.
-	imh.Errors.Push(v2.ErrorCodeUnsupported)
-	w.WriteHeader(http.StatusBadRequest)
+	imh.Errors = append(imh.Errors, errcode.NewError(v2.ErrorCodeUnsupported))
```

Comment:
can you elaborate on this one?  Are you saying we should try to get rid of the NewError() func?
When an error is generated any of them could have a `Details` section so unless we want to support ErrCode's and Errors both being used interchangeably, I think we need it.